pvlib/pvsystem.py

```diff
@@ -378,6 +378,71 @@ def get_iam(self, aoi, iam_model='physical'):
         return tuple(array.get_iam(aoi, iam_model)
                      for array, aoi in zip(self.arrays, aoi))
 
+    @_unwrap_single_value
+    def get_cell_temperature(self, poa_global, temp_air, wind_speed, model,
+                             **kwargs):
+        """
+        Determine cell temperature using the method specified by ``model``.
+
+        Parameters
+        ----------
+        poa_global : numeric or tuple of numeric
+            Total incident irradiance in W/m^2.
+
+        temp_air : numeric or tuple of numeric
+            Ambient dry bulb temperature in degrees C.
+
+        wind_speed : numeric or tuple of numeric
+            Wind speed in m/s.
+
+        model : str
+            Supported models include ``'sapm'``, ``'pvsyst'``,
+            ``'faiman'``, ``'fuentes'``, and ``'noct_sam'``
+
+        **kwargs
+            Extra arguments passed to the model function.
+
+        Returns
+        -------
+        numeric or tuple of numeric
+            values in degrees C.
+
+        See Also
+        --------
+        Array.get_cell_temperature
+
+        Notes
+        -----
+        The `temp_air` and `wind_speed` parameters may be passed as tuples
+        to provide different values for each Array in the system. If passed as
+        a tuple the length must be the same as the number of Arrays. If not
+        passed as a tuple then the same value is used for each Array.
+        """
+        poa_global = self._validate_per_array(poa_global)
+        temp_air = self._validate_per_array(temp_air, system_wide=True)
+        wind_speed = self._validate_per_array(wind_speed, system_wide=True)
+
+        # additional model-specific (and array-specific) inputs
+        extra_inputs = [{}] * self.num_arrays
+
+        if 'effective_irradiance' in kwargs:
```

Comment:
Is there a reason we cannot add `effective_irradiance=None` to the method signature and drop `kwargs`? I don't see one, but I feel like I might be forgetting something from a previous discussion.

Reply:
I think I was anticipating new models with new weather inputs, but better to cross that bridge if we come to it.  The code is much simpler now without this.